File: RetroTrack.Domain/DTOs/Controllers/Games/Responses/GetGamesForConsoleResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RetroTrack.Domain.DTOs.Controllers.Games.Responses
{
    public class GetGamesForConsoleResponse
    {
        public required int TotalCount { get; set; }
        public required int TotalPages { get; set; }
        public required ConsoleGames[] Games { get; set; }
        public required string ConsoleName { get; set; }
    }

    public class ConsoleGames
    {
        public required int GameId { get; set; }
        public required string GameTitle { get; set; }
        public required string GameGenre { get; set; }
        public required int AchievementCount { get; set; }
        public required int PlayerCount { get; set; }
        public required string GameImageUrl { get; set; }
        public required long Points { get; set; }
        public string? ConsoleName { get; set; }
    }
}

File: retrotrack.web/interfaces/api/gamess/GetGamesForConsoleResponse.ts
export interface GetGamesForConsoleResponse {
    totalCount: number
    totalPages: number
    games: Game[]
    consoleName: string
}

export interface Game {
    gameId: number
    gameTitle: string
    gameGenre: string
    achievementCount: number
    playerCount: number
    gameImageUrl: string
    points: number
    consoleName?: string // Optional, only present if consoleId is -1
}
